webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPNodeRepository.scala

```diff
@@ -189,7 +189,8 @@ class WoLDAPNodeRepository(
     val entry = mapper.nodeToEntry(node)
     for {
       con <- ldap
-      _ <- con.save(entry).chainError(s"Cannot create node with id '${node.id.value}'")
+      _   <- con.save(entry).chainError(s"Cannot create node with id '${node.id.value}'")
+      _   <- actionLogger.savePromoteToRelay(modId,actor,node, reason)
```

Comment:
I should create the eventlog, on the plugin side in `ScaleOutRelatService:promoteToRelay` (but it's an object so I have to transform it to a class to pass a eventlog service ?)

Reply:
you can't put it here, because the semantic of create node has nothing to do with promoting node to relay, it's the opposite: create node is a part of node promotion. 
Yes, change to a class and instanciate it in module conf class.